### PR TITLE
Naming and linting improvements

### PR DESCRIPTION
**Pull request by thisjunewest**
This fully lints `java-test` with the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html).

Also rewrites many class and variable names to actually makes sense, and separates out sensible classes.

Sorry @jgwilson42, it does produce more tests than before, because I removed a spurious `println` statement

### CODE REVIEW

**Review by rjmunro — APPROVED** [review]
I approve this for merging into develop. I don't think we need to wait for James until we merge into master and call this a new release.

**Review by jgwilson42 — APPROVED** [review]
Hi Pete,

Looks fine to me, let me know when you want to merge this into master and I will prep the required change to the quality-assurance repo.

Thanks,
James

**Pull request merged**
